internal/allocator/allocator.go

```diff
@@ -54,6 +54,60 @@ type alloc struct {
 	key
 }
 
+// ipPool represents a mapping of IP families (IPv4, IPv6, or both) to their respective IP addresses.
```

Comment:
this deserves a separate file I think. Also, we should think about the naming because it's not a "pool" and it's easy to mismatch it to the other "pool of ips" (and I know naming is hard :-) ).
Let's call this "allocation"